### ISSUE TITLE
XCode 10 GM and Swift 4.2 fails to build

### ISSUE THREAD

**Issue by jlaws** [Yummypets/YPImagePicker#214]
**Describe the bug**
The pod does not compile with Xcode 10 GM and Swift 4.2

**To Reproduce**
Download XCode 10 GM and try to build the library with Swift 4.2

**Expected behavior**
Build with no errors or warnings

**Screenshots**
n/a

**Environment (please complete the following information):**
 - Device: Simulators
 - OS: iOS 11 + iOS 12
 - Xcode Version 10 GM
 - Swift Version 4.2

**Installation Type**
 - Cocoapods

**Additional context**
```
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/Extensions/AVAsset+Extensions.swift:15:50: Missing argument labels 'start:end:' in call
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/Extensions/AVFoundation+Extensions.swift:26:37: Missing argument label 'preferredTimescale:' in call
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/Extensions/AVFoundation+Extensions.swift:27:32: Missing argument labels 'value:timescale:' in call
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/Extensions/AVMutableComposition+Extensions.swift:24:44: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/YPLoaders.swift:14:23: 'init(activityIndicatorStyle:)' has been renamed to 'init(style:)'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIActivityIndicatorView:6:12: 'init(activityIndicatorStyle:)' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/Extensions/UIButton+Extensions.swift:12:57: 'UIControlState' has been renamed to 'UIControl.State'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIControlState:2:18: 'UIControlState' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Models/YPPermissionDeniedPopup.swift:19:34: 'UIAlertActionStyle' has been renamed to 'UIAlertAction.Style'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIAlertActionStyle:3:18: 'UIAlertActionStyle' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Models/YPPermissionDeniedPopup.swift:28:71: 'UIApplicationOpenSettingsURLString' has been renamed to 'UIApplication.openSettingsURLString'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIApplicationOpenSettingsURLString:3:12: 'UIApplicationOpenSettingsURLString' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Models/YPPermissionDeniedPopup.swift:30:74: 'UIApplicationOpenSettingsURLString' has been renamed to 'UIApplication.openSettingsURLString'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIApplicationOpenSettingsURLString:3:12: 'UIApplicationOpenSettingsURLString' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Filters/YPPhotoFiltersVC.swift:74:67: 'UICollectionViewScrollPosition' has been renamed to 'UICollectionView.ScrollPosition'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UICollectionViewScrollPosition:2:18: 'UICollectionViewScrollPosition' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/LibraryMediaManager.swift:76:53: Missing argument labels 'start:duration:' in call
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/LibraryMediaManager.swift:107:97: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/LibraryMediaManager.swift:108:55: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/LibraryMediaManager.swift:116:60: Missing argument labels 'value:timescale:' in call
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/LibraryMediaManager.swift:76:54: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/LibraryMediaManager.swift:96:95: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/LibraryMediaManager.swift:97:95: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/YPHelper.swift:53:61: 'UIViewAnimationOptions' has been renamed to 'UIView.AnimationOptions'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIViewAnimationOptions:3:18: 'UIViewAnimationOptions' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/YPImagePicker.swift:82:69: 'kCAMediaTimingFunctionEaseInEaseOut' has been renamed to 'CAMediaTimingFunctionName.easeInEaseOut'
/Users/joelaws/Workspace/kratos/Pods/QuartzCore.kCAMediaTimingFunctionEaseInEaseOut:3:12: 'kCAMediaTimingFunctionEaseInEaseOut' was obsoleted in Swift 3
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/YPImagePicker.swift:83:31: 'kCATransitionFade' has been renamed to 'CATransitionType.fade'
/Users/joelaws/Workspace/kratos/Pods/QuartzCore.kCATransitionFade:3:12: 'kCATransitionFade' was obsoleted in Swift 3
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/SelectionsGallery/YPSelectionsGalleryVC.swift:123:15: 'UIScrollViewDecelerationRateFast' has been renamed to 'UIScrollView.DecelerationRate.fast'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIScrollViewDecelerationRateFast:3:12: 'UIScrollViewDecelerationRateFast' was obsoleted in Swift 3
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/SelectionsGallery/YPSelectionsGalleryVC.swift:124:15: 'UIScrollViewDecelerationRateNormal' has been renamed to 'UIScrollView.DecelerationRate.normal'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIScrollViewDecelerationRateNormal:3:12: 'UIScrollViewDecelerationRateNormal' was obsoleted in Swift 3
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPAlbumCell.swift:20:26: 'UITableViewCellStyle' has been renamed to 'UITableViewCell.CellStyle'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UITableViewCellStyle:2:18: 'UITableViewCellStyle' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPAlbumCell.swift:20:14: Initializer does not override a designated initializer from its superclass
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPAlbumVC.swift:67:33: 'UITableViewAutomaticDimension' has been renamed to 'UITableView.automaticDimension'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UITableViewAutomaticDimension:3:12: 'UITableViewAutomaticDimension' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPAlbumView.swift:15:19: 'init(activityIndicatorStyle:)' has been renamed to 'init(style:)'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIActivityIndicatorView:6:12: 'init(activityIndicatorStyle:)' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Filters/Video/YPVideoFiltersVC.swift:124:70: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Filters/Video/YPVideoFiltersVC.swift:233:52: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Filters/Video/YPVideoFiltersVC.swift:234:51: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Filters/Video/YPVideoFiltersVC.swift:243:64: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Filters/Video/YPVideoFiltersVC.swift:252:64: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Video/YPVideoHelper.swift:197:38: Missing argument label 'preferredTimescale:' in call
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Filters/Video/YPVideoView.swift:78:25: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Filters/Video/YPVideoView.swift:127:25: 'kCMTimeZero' has been renamed to 'CMTime.zero'
/Users/joelaws/Workspace/kratos/Pods/CoreMedia.kCMTimeZero:3:12: 'kCMTimeZero' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/YPAlerts.swift:18:75: 'UIAlertActionStyle' has been renamed to 'UIAlertAction.Style'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIAlertActionStyle:3:18: 'UIAlertActionStyle' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Helpers/YPAlerts.swift:28:75: 'UIAlertActionStyle' has been renamed to 'UIAlertAction.Style'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIAlertActionStyle:3:18: 'UIAlertActionStyle' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPAssetViewContainer.swift:25:27: 'init(activityIndicatorStyle:)' has been renamed to 'init(style:)'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIActivityIndicatorView:6:12: 'init(activityIndicatorStyle:)' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/BottomPager/YPBottomPager.swift:55:15: 'willMove(toParentViewController:)' has been renamed to 'willMove(toParent:)'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIViewController:12:15: 'willMove(toParentViewController:)' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/BottomPager/YPBottomPager.swift:56:13: 'addChildViewController' has been renamed to 'addChild(_:)'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIViewController:11:15: 'addChildViewController' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/BottomPager/YPBottomPager.swift:59:15: 'didMove(toParentViewController:)' has been renamed to 'didMove(toParent:)'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIViewController:20:15: 'didMove(toParentViewController:)' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPLibraryVC+PanGesture.swift:72:28: 'UIGestureRecognizerState' has been renamed to 'UIGestureRecognizer.State'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIGestureRecognizerState:2:18: 'UIGestureRecognizerState' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPLibraryVC+PanGesture.swift:97:35: 'UIGestureRecognizerState' has been renamed to 'UIGestureRecognizer.State'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIGestureRecognizerState:2:18: 'UIGestureRecognizerState' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPLibraryVC+PanGesture.swift:116:32: 'UIGestureRecognizerState' has been renamed to 'UIGestureRecognizer.State'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UIGestureRecognizerState:2:18: 'UIGestureRecognizerState' was obsoleted in Swift 4.2
/Users/joelaws/Workspace/kratos/Pods/YPImagePicker/Source/Pages/Gallery/YPLibraryVC.swift:241:62: 'UICollectionViewScrollPosition' has been renamed to 'UICollectionView.ScrollPosition'
/Users/joelaws/Workspace/kratos/Pods/UIKit.UICollectionViewScrollPosition:2:18: 'UICollectionViewScrollPosition' was obsoleted in Swift 4.2
```

**Comment by s4cha** [comment]
Looking at it at the moment :)

**Comment by s4cha** [comment]
This is being addressed here https://github.com/Yummypets/YPImagePicker/pull/222.
Until it's merged to mater, you can point to `Xcode10` branch :)

**Comment by s4cha** [comment]
@jlaws We are tracking this in here as well : https://github.com/Yummypets/YPImagePicker/issues/224
The last bit to solve comes from one of our dependency so I hope they fix it asap so that we can rollout a new version.
Thanks for being so patient 🙏

**Comment by jlaws** [comment]
Thanks for the update!  Just need this and Stevia to update and all my pods will be on Swift 4.2

**Comment by s4cha** [comment]
@jlaws 
This should be fixed by now, 3.5.1 is available on Cocoapods 🚀 
https://github.com/Yummypets/YPImagePicker/releases/tag/3.5.1
http://cocoapods.org/pods/YPImagePicker

Let me know if that works,
Thanks a ton for your patience :)
Cheers!

**Comment by jlaws** [comment]
Works great, thank you!
